lib/Search/BasicSearch/BasicSearchEngine.php

```diff
@@ -0,0 +1,526 @@
+            if (file_exists('custom/modules/' . $moduleName . '/metadata/metafiles.php')) {
+                require('custom/modules/' . $moduleName . '/metadata/metafiles.php');
+            } elseif (file_exists('modules/' . $moduleName . '/metadata/metafiles.php')) {
+                require('modules/' . $moduleName . '/metadata/metafiles.php');
+            }
+
+
+            if (!empty($metafiles[$moduleName]['searchfields'])) {
+                /** @noinspection PhpIncludeInspection */
+                require $metafiles[$moduleName]['searchfields'];
+            } elseif (is_file("modules/{$moduleName}/metadata/SearchFields.php")) {
+                require "modules/{$moduleName}/metadata/SearchFields.php";
+            }
+
+            //Load custom SearchFields.php if it exists
+            if (is_file("custom/modules/{$moduleName}/metadata/SearchFields.php")) {
+                require "custom/modules/{$moduleName}/metadata/SearchFields.php";
+            }
+
+            // If there are $searchFields are empty, just continue, there are no search fields defined for the module
+            if (empty($searchFields[$moduleName])) {
+                continue;
+            }
+
+            $isCustomModule = preg_match('/^([a-z0-9]{1,5})_([a-z0-9_]+)$/i', $moduleName);
+
+            // If the bean supports unified search or if it's a custom module bean and unified search is not defined
+            if (!empty($dictionary[$beanName]['unified_search']) || $isCustomModule) {
+                $fields = [];
+                foreach ($dictionary [$beanName]['fields'] as $field => $def) {
+                    // We cannot enable or disable unified_search for email in the vardefs as we don't actually have a vardef entry for 'email'
+                    // the searchFields entry for 'email' doesn't correspond to any vardef entry. Instead it contains SQL to directly perform the search.
+                    // So as a proxy we allow any field in the vardefs that has a name starting with 'email...' to be tagged with the 'unified_search' parameter
+
+                    if (str_contains($field, 'email')) {
+                        $field = 'email';
+                    }
+
+                    if (str_contains($field, 'phone')) {
+                        $field = 'phone';
+                    }
+
+                    if (!empty($def['unified_search']) && isset($searchFields [$moduleName] [$field])) {
+                        $fields [$field] = $searchFields [$moduleName] [$field];
+                    }
+                }
+
+                foreach ($searchFields[$moduleName] as $field => $def) {
+                    if (!empty($def['force_unifiedsearch'])) {
+                        $fields[$field] = $def;
+                    }
+                }
+
+                if (!empty($fields)) {
+                    $supported_modules [$moduleName] ['fields'] = $fields;
```

Comment:
Hi @Dillon-Brown minor formatting change needed on `$supported_modules [$moduleName] ['fields'] `.

Reply:
Pushed, strange my IDE doesn't want to fix these though 🤔

Reply:
thank you